### PR TITLE
[#IP-83] Transparent middleware that checks request's accept and content-type headers

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### List of Changes
- Fixed spaces and new lines on `launch.json`
- Created `checkHeaders.ts` file
- Created `checkHeaders.tests.ts` test file
- Changed `app.ts` to import checkAcceptHeader and checkContenttypHeader
- Changed `app.ts` to use checkAcceptHeader and checkContenttypHeader

#### Motivation and Context
We want to have a middleware that warns us about incoherent requests (e.g. the content-type header should have "application/json" if an endpoint accept a json, the accept header should be "application/json" if the client expect to receive a json).

#### Notes
- checkAcceptHeader and checkContenttypHeader middlewares are at the application level
- these middlewares are using winston.Logger to log warnings

#### How Has This Been Tested?
It has been tested by performing:
- yarn install --frozen-lockfile
- yarn build
- yarn lint
- yarn test
- some calls to io-backend api using io-mock.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/middleware/checkHeaders.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      .chain(
+        fromPredicate(
+          commonValues => commonValues.length > 0,
+          () => "unexpected-header-values"
+        )
+      )
+      .map(_ => void 0)
+  );
+
+export const checkAcceptHeader = (
```
Comment (balanza) [inline src/utils/middleware/checkHeaders.ts:53]: add a docblock comment to explain what this function does

File: src/utils/middleware/checkHeaders.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        default:
+          return "Unexpected case";
+      }
+    })
+    .mapLeft(log.warn);
+
+  next();
+};
+
+export const checkContentTypeHeader = (
```
Comment (balanza) [inline src/utils/middleware/checkHeaders.ts:78]: add a docblock comment to explain what this function does

**Review by balanza — CHANGES_REQUESTED** [review]
Code is good, we just need some more comments explaining what's going on, at least for _public methods_

File: src/utils/middleware/checkHeaders.ts (hunk 1 above)
Comment (michaeldisaro) [reply src/utils/middleware/checkHeaders.ts:59]: done

File: src/utils/middleware/checkHeaders.ts (hunk 2 above)
Comment (michaeldisaro) [reply src/utils/middleware/checkHeaders.ts:91]: done

**Comment by michaeldisaro** [comment]
@balanza I also removed a useless map to void.

File: src/utils/middleware/checkHeaders.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          // eslint-disable-next-line functional/prefer-readonly-type
+          expectedHeaderValues as string[]
+        )
+      )
+      .chain(
+        fromPredicate(
+          commonValues => commonValues.length > 0,
+          () => "unexpected-header-values"
+        )
+      )
```
Comment (michaeldisaro) [inline src/utils/middleware/checkHeaders.ts:49]: @balanza there was a .map(_ => void 0) here. I removed it and changed the right side of either to ReadonlyArray<string>.

**Pull request closed without merging**